packages/headless/src/controllers/instant-results/instant-results.ts

```diff
@@ -0,0 +1,174 @@
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {Result} from '../../api/search/search/result';
+import {InstantResultSection} from '../../state/state-sections';
+import {buildController, Controller} from '../controller/headless-controller';
+import {instantResults} from '../../app/reducers';
+import {loadReducerError} from '../../utils/errors';
+import {validateOptions} from '../../utils/validate-payload';
+import {
+  clearExpiredResults,
+  registerInstantResults,
+  updateInstantResultsQuery,
+} from '../../features/instant-results/instant-results-actions';
+import {
+  InstantResultOptions,
+  instantResultsOptionsSchema,
+} from './instant-results-options';
+import {SearchAPIErrorWithStatusCode} from '../../api/search/search-api-error-response';
+import {SerializedError} from '@reduxjs/toolkit';
+import {fetchInstantResults} from '../../features/search/search-actions';
+import {hasExpired} from '../../features/instant-results/instant-results-state';
+import {randomID} from '../../utils/utils';
+
+export interface InstantResultProps {
+  options: InstantResultOptions;
+}
+
+/**
+ * The `InstantResults` controller allows the end user to manage instant results queries.
+ */
+export interface InstantResults extends Controller {
+  /**
+   * Updates the specified query and shows instant results for it.
+   *
+   * @param q - The query to get instant results for. For more precise instant results, query suggestions are recommended.
+   */
+  updateQuery(q: string): void;
+  /**
+   * Fetches results from the server for a query.
+   */
+  updateResultsForQuery(q: string): void;
```

Comment:
Do developers need to call this function? It looks like `updateQuery` is what we expect them to use.

I would not expose this method unless necessary.